packages/discord.js/src/structures/InteractionCollector.js

```diff
@@ -141,6 +147,7 @@ class InteractionCollector extends Collector {
     if (this.interactionType && interaction.type !== this.interactionType) return null;
     if (this.componentType && interaction.componentType !== this.componentType) return null;
     if (this.messageId && interaction.message?.id !== this.messageId) return null;
+    if (this.messageInteractionId && interaction.message?.interaction?.id !== this.messageInteractionId) return null;
```

Comment:
Alright, so now that I've seen this line it's clear to me that its an interaction id, but my above comments about the description still applies